iroh/src/main.rs

```diff
@@ -25,11 +25,23 @@ fn main() -> Result<()> {
 
 async fn main_impl() -> Result<()> {
     let lp = tokio_util::task::LocalPoolHandle::new(num_cpus::get());
+    let cli = Cli::parse();
+
+    #[cfg(unix)]
+    if let Some(log_fd) = cli.log_fd {
+        use std::os::unix::io::FromRawFd;
+
+        let f = unsafe { std::fs::File::from_raw_fd(log_fd) };
```

Comment:
should this be `manuallydrop`? from `from_raw_fd` takes ownership and closes the descriptor when the `File` is dropped

Reply:
I believe the ownership is passed into the `.with_witer()` and tracing keeps it alive?